test/helpers/clean_method.ts

```diff
@@ -0,0 +1,311 @@
+
+type AddOptions = {
+	/**
+	 * @description
+	 * If `true`, the method will be executed only one time and then removed from the store.
+	 *
+	 * @default true
+	 */
+	executeOnce?: boolean
+}
+
+type ResetMethodOptions = {
+	/**
+	 * @description
+	 * If `true`, only the methods that have the option `executeOnce` set to `true` will be removed.
+	 *
+	 * @default false
+	 */
+	removeOnce?: boolean
+}
+
+type ResetOptions = {
+	method?: ResetMethodOptions
+}
+
+/**
+ * @description
+ * Class to help clean the environment or close opened connections after tests finish.
+ * Also, you can add custom methods to be executed after the tests finish, like
+ * deleting temporary files or closing connections.
+ *
+ * @example
+ * ```
+ * import { describe, it } from 'node:test'
+ * import mqtt from './src'
+ * import serverBuilder from './test/server_helpers_for_client_tests'
+ * import CleanMethod from './test/helpers/clean_method'
+ *
+ *
+ * describe('Test', () => {
+ *     const cleanMethod = new CleanMethod()
+ *
+ *     it('should clean the client and server', (t, done) => {
+ *         t.after(async () => {
+ *             await cleanMethod.closeClientAndServer()
+ *         })
+ *
+ *         const server = serverBuilder('8883')
+ *         const client = mqtt.connect('mqtt://localhost')
+ *
+ *         cleanMethod.setServer(server)
+ *         cleanMethod.setClient(client)
+ *     })
+ * })
+ * ```
+ *
+ * @example
+ * ```
+ * import { describe, it, after } from 'node:test'
+ * import mqtt from './src'
+ * import serverBuilder from './test/server_helpers_for_client_tests'
+ * import CleanMethod from './test/helpers/clean_method'
+ *
+ *
+ * describe('Test', () => {
+ *
+ *     const cleanMethod = new CleanMethod()
+ *     let server = serverBuilder('8883')
+ *
+ *     cleanMethod.add({}, async () => {
+ *         if (server?.listening) {
+ *             await new Promise<void>((resolve, reject) => {
+ *                 server.close((err) => {
+ *                     if (err) reject(err)
+ *                     else resolve()
+ *                 })
+ *             })
+ *         }
+ *     })
+ *
+ *     after(async () => {
+ *         await cleanMethod.closeAllMethods()
+ *     })
+ *
+ *     it('should clean the client and server', (t, done) => {
+ *         server = serverBuilder('8883')
+ *         const client = mqtt.connect('mqtt://localhost')
+ *
+ *         cleanMethod.setClient(client)
+ *         done()
+ *     })
+ *
+ * })
+ * ```
+ */
+class CleanMethod {
+	#client: MqttClient | null
+
+	#server: ServerBuilderInstance | null
+
+	#methods: Map<
+		string,
+		{
+			options: AddOptions
+			method: Promise<any> | ((...args: any[]) => Promise<any>)
+			args: any[]
+		}
+	>
+
+	constructor() {
+		this.#client = null
+		this.#server = null
+		this.#methods = new Map()
+	}
+
+	/**
+	 * @description
+	 * Set the client to be used as default to close.
+	 */
+	setClient(client: MqttClient | null) {
+		this.#client = client
+	}
```

Comment:
could make sense to make this throw in case there was already a client set? Just to be sure we don't call this multiple times

Reply:
I would also refactor this to simply add the closeClient method to methods instead